### DISCUSSION TITLE
Error! No Next.js version could be detected in your project. Make sure `"next"` is installed in "dependencies" or "devDependencies"

### DISCUSSION THREAD

**Discussion by ramseylove** [vercel/vercel#8396] [Help] [answered]
Migrating app hosted on verbal from CRA to nextjs. I was able to get it up and running locally and  it built successfully. When deploying to Vercel from GitHub, the build fails with this error:

```
[19:18:18.407] Cloning github.com/ramseylove/podcast-player (Branch: prod, Commit: dc738c7)
[19:18:18.779] Cloning completed: 372.096ms
[19:18:19.485] Looking up build cache...
[19:18:19.726] Build Cache not found
[19:18:19.944] Running "vercel build"
[19:18:20.434] Vercel CLI 28.0.1
[19:18:20.692] Your application is being built using `next build`. If you need to define a different build step, please create a `vercel-build` script in your `package.json` (e.g. `{ "scripts": { "vercel-build": "npm run prepare && next build" } }`).
[19:18:20.693] Installing dependencies...
[19:18:21.034] yarn install v1.22.17
[19:18:21.046] warning package.json: No license field
[19:18:21.056] info No lockfile found.
[19:18:21.060] warning No license field
[19:18:21.061] [1/4] Resolving packages...
[19:18:21.062] [2/4] Fetching packages...
[19:18:21.063] [3/4] Linking dependencies...
[19:18:21.067] [4/4] Building fresh packages...
[19:18:21.070] success Saved lockfile.
[19:18:21.073] Done in 0.04s.
[19:18:21.084] Warning: Could not identify Next.js version, ensure it is defined as a project dependency.
[19:18:21.085] Error! No Next.js version could be detected in your project. Make sure `"next"` is installed in "dependencies" or "devDependencies"
```

The only other context, is the first error I received is that my root directory was configured incorrectly, but once I set that to the /pages directory, that error went away and I received this one

package.json:
```
{
  "name": "podcast-player",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest",
      "next"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "dependencies": {
    "@emotion/react": "^11.8.2",
    "@emotion/styled": "^11.8.1",
    "@jy95/material-ui-image": "^4.0.0",
    "@mui/icons-material": "^5.5.0",
    "@mui/material": "^5.5.0",
    "@testing-library/jest-dom": "^5.16.2",
    "@testing-library/react": "^12.1.3",
    "@testing-library/user-event": "^13.5.0",
    "arraybuffer-xml-parser": "^0.5.0",
    "next": "^12.2.5",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-icons": "^4.3.1",
    "svgo": "^2.8.0",
    "web-vitals": "^2.1.4"
  },
  "devDependencies": {
    "eslint": "^8.21.0",
    "eslint-config-next": "^12.2.5",
    "eslint-config-react-app": "^7.0.1",
    "json-server": "^0.17.0"
  }
}
```

Next is definitely in my dependencies and in my lock file. Not sure what else I can do to troubleshoot. Let me know if there is any other information I can provide that might help.

**Comment by amyegan** [comment]
Thanks for reaching out @ramseylove. Can you share your Project Settings? If this was previously deployed as CRA, then the settings might need to be adjusted for a Next.js build. For example, this is what I have for a very basic Next app example:

![image](https://user-images.githubusercontent.com/8247105/184692534-a03b8235-8cf3-477a-a012-c6cdd1cc6c8d.png)

**Reply by ramseylove** [reply]
The only thing I see different is the root directory. I I do not recall the error exactly but it was basically stating that it couldn't find a entry point for the application. When I set it too pages, that error went away and the build process got further but I then received this error.

<img width="960" alt="CleanShot 2022-08-15 at 14 35 20@2x" src="https://user-images.githubusercontent.com/3958672/184695665-deb64c16-2378-4fe5-b166-3c52df84fb45.png">

**Reply by amyegan** [reply]
Is this a monorepo, or a single app repo?

**Comment by ramseylove** [comment, marked as answer]
Single app repo
________________________________

**Reply by amyegan** [reply]
In that case, I wouldn't expect `pages` to be the correct Root Directory setting. Is your repo public, or can you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)?

**Reply by ramseylove** [reply]
You got me thinking, thank you. Originally with it being configured with CRA, the "include source files outside the root directory" option was not checked for me. So I removed pages as the root directory and just left the option checked and it built just fine. Thank you for the help.

**Comment by productdevbook** [comment]
![image](https://user-images.githubusercontent.com/38668796/189815686-fab8f33a-c25e-4e71-8eb4-9b835279f245.png)


vercel cli is problem, but auto vercel dont problem some root setting 🤔

![image](https://user-images.githubusercontent.com/38668796/189815712-b5512d10-b28c-4640-9f6e-06eae7cdb1fc.png)

cc: @amyegan

**Reply by amyegan** [reply]
Hi @productdevbook! The original problem for this discussion was already solved. Can you open a [new discussion](https://github.com/vercel/community/discussions/new) and provide more detail about your project? It looks like you have a monorepo, but there are a lot of different tools and options that I can't guess from this post. If you can also provide a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) or share your repo, that would be helpful for debugging.

_New discussion: https://github.com/orgs/vercel/discussions/851_
